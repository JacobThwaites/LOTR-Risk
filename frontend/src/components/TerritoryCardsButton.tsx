import React from "react";
import BlankCards from "../assets/blank-cards.svg";
import CustomButton from "./common/CustomButton";

export default function TerritoryCardsButton(props: { onClick: any, numCards: number }): JSX.Element {
    const label = (
        <>
            <img id='show-territory-cards-button--img' src={BlankCards} alt="asdf"/>
            <div id='show-territory-cards-button--num-cards'>{props.numCards}</div>
        </>
    )
    return (
        <CustomButton id='show-territory-cards-button' onClick={props.onClick} label={label} title='Territory Cards'/>
    )
}