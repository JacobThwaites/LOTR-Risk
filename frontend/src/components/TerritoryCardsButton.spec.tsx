import '@testing-library/jest-dom'
import React from 'react';
import { render } from '@testing-library/react'
import TerritoryCardsButton from './TerritoryCardsButton';

it('renders TerritoryCardsButton correctly', async () => {
    const numCards = 2;
    const { container } = render(
        <TerritoryCardsButton 
            onClick={() => {}}
            numCards={numCards}
        />
    );
    expect(container).toMatchSnapshot();

    const numCardsDiv = container.getElementsByClassName('show-territory-cards-button--num-cards')[0];
    expect(numCardsDiv).toHaveTextContent(`${numCards}`);
});