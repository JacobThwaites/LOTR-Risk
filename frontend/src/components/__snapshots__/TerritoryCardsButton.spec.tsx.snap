// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders TerritoryCardsButton correctly 1`] = `
<div>
  <button
    class="custom-button"
    data-testid="show-territory-cards-button"
    id="show-territory-cards-button"
    title="Territory Cards"
  >
    <img
      alt="Territory Cards"
      id="show-territory-cards-button--img"
      src="blank-cards.svg"
    />
    <div
      class="show-territory-cards-button--num-cards"
    >
      2
    </div>
  </button>
</div>
`;
